### PR TITLE
Validate FHIR version for Dicom-Cast

### PR DESCRIPTION
**Pull request by abdullah248**
## Description
Update code to validate that FHIR server is version R4 and also update docs to mention R4 requirement in order for Dicom-Cast to work with FHIR.

## Related issues
[AB#76995](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/76995)

## Testing
Added unit tests and manually tested with correct version of FHIR.

### CODE REVIEW

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs [hunk 1]
```diff
@@ -51,6 +57,30 @@ public Task<ImagingStudy> RetrieveImagingStudyAsync(Identifier identifier, Cance
         public Task<Endpoint> RetrieveEndpointAsync(string queryParameter, CancellationToken cancellationToken)
             => SearchByQueryParameterAsync<Endpoint>(queryParameter, cancellationToken);
 
+        /// <inheritdoc/>
+        public async System.Threading.Tasks.Task ValidateFhirService(CancellationToken cancellationToken)
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs:61]: nit: CheckFhirServiceCapability?

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs [hunk 2]
```diff
@@ -51,6 +57,30 @@ public Task<ImagingStudy> RetrieveImagingStudyAsync(Identifier identifier, Cance
...
+        {
+            using FhirResponse<CapabilityStatement> response = await _fhirClient.ReadAsync<CapabilityStatement>("metadata", cancellationToken);
+            var version = response.Resource.FhirVersion ?? throw new InvalidFhirServerException("FHIR server version cannot be validated, should be R4");
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs:64]: We should create a custom exception with strings in resx file. Is that not what we are folowing dicom-cast?

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs [hunk 3]
```diff
@@ -51,6 +57,30 @@ public Task<ImagingStudy> RetrieveImagingStudyAsync(Identifier identifier, Cance
         public Task<Endpoint> RetrieveEndpointAsync(string queryParameter, CancellationToken cancellationToken)
             => SearchByQueryParameterAsync<Endpoint>(queryParameter, cancellationToken);
 
+        /// <inheritdoc/>
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs:60]: nit using System.Threading.Tasks; and remove the full name

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs (hunk 1 above)
Comment (abdullah248) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs]: Updated

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs (hunk 2 above)
Comment (abdullah248) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs]: I added strings to the resx file. The exception thrown is a custom exception I added.

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs (hunk 3 above)
Comment (abdullah248) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Fhir/FhirService.cs:61]: Updated, thanks!

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/DicomCastCoreResource.resx [hunk 4]
```diff
@@ -120,6 +120,12 @@
   <data name="DataStoreOperationFailed" xml:space="preserve">
     <value>The data store operation failed.</value>
   </data>
+  <data name="FailedToValidateFhirVersion" xml:space="preserve">
+    <value>FHIR server version cannot be validated, should be R4</value>
+  </data>
+  <data name="FhirServerTransactionNotSupported" xml:space="preserve">
+    <value>FHIR server does not support transactions</value>
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/DicomCastCoreResource.resx:127]: nit: exceptions should provide details on how to fix it if possible. For this we could change it to.
"Configured FHIR service does not support transactions need to publish Fhir changes. Configure a FHIR service with transaction capability like FHIR-server"

Comment (abdullah248) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/DicomCastCoreResource.resx]: Updated, thanks!

**Review by smithago — APPROVED** [review]

**Pull request merged**